charmcraft/env.py

```diff
@@ -86,3 +86,38 @@ def is_charmcraft_running_in_managed_mode():
     """Check if charmcraft is running in a managed environment."""
     managed_flag = os.getenv("CHARMCRAFT_MANAGED_MODE", "n")
     return distutils.util.strtobool(managed_flag) == 1
+
+
+@dataclasses.dataclass(frozen=True)
+class CharmhubConfig:
+    """Definition of Charmhub endpoint configuration."""
+
+    api_url: str = "https://api.charmhub.io"
+    storage_url: str = "https://storage.snapcraftcontent.com"
+    registry_url: str = "https://registry.jujucharms.com"
+
+
+DEFAULT_CHARMHUB_CONFIG = CharmhubConfig()
+STAGING_CHARMHUB_CONFIG = CharmhubConfig(
+    api_url="https://api.staging.charmhub.io",
+    storage_url="https://storage.staging.snapcraftcontent.com",
+    registry_url="https://registry.staging.jujucharms.com",
+)
+
+
+def get_store_config() -> CharmhubConfig:
+    """Get the appropriate configuration for the store."""
+    config_str = os.getenv("CHARMCRAFT_STORE_CONFIG", "")
+    if not config_str:
+        return DEFAULT_CHARMHUB_CONFIG
+    if config_str.lower() == "staging":
+        return STAGING_CHARMHUB_CONFIG
```

Comment:
This is based on conversation with @sergiusens yesterday - we're going to get rid of that config option for craft-application stuff. This will be a major release anyway and it shouldn't be used outside of our team and some internal tests on other teams, so it shouldn't be a big issue.

Reply:
https://github.com/canonical/charmcraft/issues/1359